### PR TITLE
Add Power Requested trigger

### PR DESCRIPTION
**Pull request by therealfarfetchd**
Trigger turns on when requested power is more than 1 MJ.

### CODE REVIEW

**Comment by AlexIIL** [comment]
Erm, can you revert your cleanup comment and just check to see if the requested is greater than 0 for any of them? (As some things request less than 1 MJ I think?)

Although you could probably extract the implementation of `PipeFlowPower.Section.getPowerRequested()` to the main class and provide the `this.side` as null for the trigger, then check and cast the `IPipeHolder.getFlow()` into `PipeFlowPower` rather than the `TilePipeHolder` itself.

(Generally we've found replacing lambdas with normal loops to be quite a decent optimisation)

**Comment by therealfarfetchd** [comment]
The cleanup commit fixes a bug, but I guess I can convert that into a normal loop.

**Comment by therealfarfetchd** [comment]
At first, it cycled through the facings one by one and when it found a capability, it took it. Now if the only power consumer was on that side, getPowerRequested() would return 0 since it filters out the requested power of the side you got the capability from. That is not an issue anymore though

File: common/buildcraft/transport/statements/TriggerPowerRequested.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+public class TriggerPowerRequested extends BCStatement implements ITriggerInternal {
+
+    public TriggerPowerRequested() {
+        super("buildcraft:powerRequested");
+    }
+
+    @Override
+    public boolean isTriggerActive(IStatementContainer source, IStatementParameter[] parameters) {
+        final PipeFlowPower flow = (PipeFlowPower) ((IPipeHolder) source.getTile()).getPipe().getFlow();
```
Comment (AlexIIL) [inline common/buildcraft/transport/statements/TriggerPowerRequested.java]: Sorry about my last bit of advice, clearly I had forgotten how triggers worked :/

```suggestion
        if (!(source instanceof IGate)) {
            return false;
        }
        IPipeFlow f = ((IGate) source).getPipeHolder().getPipe().getFlow();
        if (!(f instanceof PipeFlowPower)) {
            return false;
        }
        final PipeFlowPower flow = (PipeFlowPower) f;
```

**Comment by AlexIIL** [comment]
> build failed with import order violation

Oh well obviously I can't merge this yet then :p

**Comment by AlexIIL** [comment]
Thanks!

**Pull request merged**